ngraph/src/ngraph/frontend/onnx_import/onnx.cpp

```diff
@@ -43,17 +43,50 @@ namespace ngraph
 
                 struct stream_parse : ngraph_error
                 {
-                    explicit stream_parse(std::istream&)
-                        : ngraph_error{"Failure parsing data from the provided input stream"}
+                    explicit stream_parse(const std::string& stream_type)
+                        : ngraph_error{"Failure parsing data from the provided input " +
+                                       stream_type + "."}
+                    {
+                    }
+                };
+
+                struct stream_corrupted : ngraph_error
+                {
+                    explicit stream_corrupted()
+                        : ngraph_error{"Provided input stream is in incorrect state."}
                     {
                     }
                 };
 
             } // namespace error
-        }     // namespace detail
+
+            std::shared_ptr<Function> get_ng_function(const ONNX_NAMESPACE::ModelProto& model_proto)
```

Comment:
do we really need it to be a function if it is called once?

Reply:
For me it's just more clear logic. I've been testing multiple solutions with multiple calls to this function. However at least for know it is called once.